New unicity policies for feeds with bad GUIDs

Alkarex opened the pull request:
fix https://github.com/FreshRSS/FreshRSS/issues/4485
fix https://github.com/FreshRSS/FreshRSS/issues/5210
fix https://github.com/FreshRSS/FreshRSS/issues/5257
fix https://github.com/FreshRSS/FreshRSS/issues/5410
fix https://github.com/FreshRSS/FreshRSS/issues/5909
fix https://github.com/FreshRSS/FreshRSS/issues/6658

Alkarex committed:
Use hash of link+title fallback for bad GUIDs

Frenzie reviewed (approved):
Maybe there's another one we should toss in there as well, like published?

math-GH committed:
Merge branch 'edge' into better-bad-guids

eht16 commented:
I would second the idea to include "published" and/or some aliases of it.

Recently, I stumbled upon https://status.otc-service.com/rss/?mt=Europe%20%28Germany%29 which does not have a "guid" and the links in there are *not* unique, only the "time" field seems usable to distinguish the posts.

Alkarex committed:
Merge branch 'edge' into better-bad-guids

Alkarex committed:
Merge branch 'edge' into better-bad-guids

Alkarex committed:
Merge branch 'edge' into better-bad-guids

Alkarex commented:
Let's first make a PR to add a new feed-level option to override the ID logic (e.g. force use of `link`)

Alkarex committed:
Minor syntax

Alkarex committed:
Merge branch 'edge' into better-bad-guids

Alkarex committed:
Add decideGuid

Alkarex committed:
Simplify _guid()

Alkarex committed:
Default fallback

Alkarex committed:
Draft of UI

Alkarex committed:
Rename to unicityPolicy + i18n

Alkarex commented:
![image](https://github.com/user-attachments/assets/3cad7f4a-34e8-4e15-86d1-2bb3b6484bc5)

Alkarex commented:
Example of logs in case of invalid IDs:

![image](https://github.com/user-attachments/assets/45585f0f-8c41-4655-8893-489063bc2a77)

Alkarex committed:
A couple of fixes

Alkarex committed:
degraded

Alkarex committed:
Help

Frenzie commented on app/i18n/en/sub.php:
```diff
@@ -221,6 +221,14 @@
 		'title_add' => 'Add an RSS feed',
 		'ttl' => 'Do not automatically refresh more often than',
+		'unicityPolicy' => array(
+			'_' => 'Articles unicity',
```
Maybe something like:
```suggestion
			'_' => 'Article uniqueness criteria',
```
Or:
```suggestion
			'_' => 'Article duplicate matching criteria',
```

Which is a long way of saying perhaps `unicity` isn't the most common word.

Alkarex replied:
Interesting ChatGPT's feedback 🤖
![image](https://github.com/user-attachments/assets/2961bba6-dded-4121-8ce3-7d5f7dbb7a56)

Indeed not the most common word but not so rare either, is it ? And probably "guessable". I do not really mind, but *unicity* is slightly more precise here.

It is important to be understandable indeed. On the other hand, this option should not be too tempting to fiddle with :-P

Let's see what our testers say, if any

Frenzie replied:
Either way I would make it article (singular) unicity. ^_^

Alkarex committed:
Slight rewording

Alkarex committed:
Rename variables to unicityCriteria

Alkarex committed:
Merge branch 'edge' into better-bad-guids

Alkarex committed:
Change automatic criteria degrading logic
Do not fall back to hash of content automatically, as this seems to often create too many duplicates

Alkarex committed:
Typo

Alkarex committed:
unicityCriteriaForced

Alkarex committed:
label title

Alkarex committed:
Merge branch 'edge' into better-bad-guids

Alkarex committed:
invalidGuidsTolerance

Alkarex commented:
I have observed several examples of feeds with temporary invalid GUIDs (non-unique) due to e.g. temporary duplicated articles.
This leads to premature changes of unicity criteria and unwanted duplicates.

So here is a new tolerance parameter. I will test some more.

> `$invalidGuidsTolerance` (default 0.05) The maximum ratio (rounded) of invalid GUIDs to tolerate before degrading the unicity criteria.
> Example for 0.05 (5% rounded): tolerate 0 invalid GUIDs for up to 9 articles, 1 for 10, 2 for 30, 3 for 50, 4 for 70, 5 for 90, 6 for 110, etc.
> The default value of 5% rounded was chosen to allow 1 invalid GUID for feeds of 10 articles, which is a frequently observed amount of articles.

Alkarex committed:
Forgotten parameter

Frenzie commented:
Temporarily duplicated? Interesting. Sounds good! 👍

Alkarex commented:
Example right now https://www.academie-sciences.fr/fr/?option=com_obrss&task=feed&id=2:fil-d-actualite-de-l-academie-des-sciences&format=feed

Error on a single entry `http://www.academie-sciences.fr/fr/Communiques-de-presse/communique-de-presse-prix-nobel-physiologie-ou-medecine.html` out of 30

<details>
<summary>feed.xml</summary>

```xml
<?xml version="1.0" encoding="utf-8"?>
<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:content="http://purl.org/rss/1.0/modules/content/" xmlns:media="http://search.yahoo.com/mrss/">
	<channel>
		<title>Les derniers articles publiés sur le site de l'Académie des sciences</title>
		<description></description>
		<link>https://www.academie-sciences.fr/</link>
		<lastBuildDate>2024-10-12T09:40:14+02:00</lastBuildDate>
		<generator>FeedCreator 1.8.3 (obRSS 3.4.2)</generator>
		<atom:link href="https://www.academie-sciences.fr/fr/?option=com_obrss&amp;task=feed&amp;id=2:fil-d-actualite-de-l-academie-des-sciences&amp;format=feed" rel="self" type="application/rss+xml" />
		<language>fr-FR</language>
		<item>
			<title>Communiqué de presse | Prix Nobel Physiologie ou médecine</title>
			<link>http://www.academie-sciences.fr/fr/Communiques-de-presse/communique-de-presse-prix-nobel-physiologie-ou-medecine.html</link>
			<description></description>
			<pubDate>2024-10-11T12:06:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Communiques-de-presse/communique-de-presse-prix-nobel-physiologie-ou-medecine.html</guid>
		</item>
		<item>
			<title>Communiqué de presse | Prix Nobel Chimie</title>
			<link>http://www.academie-sciences.fr/fr/Communiques-de-presse/communique-de-presse-prix-nobel-physiologie-ou-medecine.html</link>
			<description></description>
			<pubDate>2024-10-10T12:06:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Communiques-de-presse/communique-de-presse-prix-nobel-physiologie-ou-medecine.html</guid>
		</item>
		<item>
			<title>Affleck Ian</title>
			<link>http://www.academie-sciences.fr/fr/In-memoriam/affleck-ian.html</link>
			<description></description>
			<pubDate>2024-10-04T13:00:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/In-memoriam/affleck-ian.html</guid>
		</item>
		<item>
			<title>Ghislain de Marsily</title>
			<link>http://www.academie-sciences.fr/fr/In-memoriam/ghislain-de-marsily.html</link>
			<description></description>
			<pubDate>2024-04-21T14:47:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/In-memoriam/ghislain-de-marsily.html</guid>
		</item>
		<item>
			<title>Communiqué de presse | Prix Nobel de Physique </title>
			<link>http://www.academie-sciences.fr/fr/Communiques-de-presse/communique-de-presse-prix-nobel-de-physique.html</link>
			<description></description>
			<pubDate>2024-10-10T11:45:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Communiques-de-presse/communique-de-presse-prix-nobel-de-physique.html</guid>
		</item>
		<item>
			<title>Rencontre avec Jocelyne Troccaz - Des automates aux robots chirurgicaux</title>
			<link>http://www.academie-sciences.fr/fr/Colloques-conferences-et-debats/rencontre-avec-jocelyne-troccaz-des-automates-aux-robots-chirurgicaux.html</link>
			<description></description>
			<pubDate>2024-11-05T13:06:00+01:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Colloques-conferences-et-debats/rencontre-avec-jocelyne-troccaz-des-automates-aux-robots-chirurgicaux.html</guid>
		</item>
		<item>
			<title>Rencontre avec Bérengère Dubrulle  - Les mystères de la turbulence - Cycle Rencontre à l'Académie des sciences</title>
			<link>http://www.academie-sciences.fr/fr/Seances-publiques/rencontre-avec-berengere-dubrulle-les-mysteres-de-la-turbulence-cycle-rencontre-a-l-academie-des-sciences.html</link>
			<description></description>
			<pubDate>2024-10-29T10:40:00+01:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Seances-publiques/rencontre-avec-berengere-dubrulle-les-mysteres-de-la-turbulence-cycle-rencontre-a-l-academie-des-sciences.html</guid>
		</item>
		<item>
			<title>Une Babel des sciences ? Traductions et sciences ouvertes </title>
			<link>http://www.academie-sciences.fr/fr/Colloques-conferences-et-debats/une-babel-des-sciences-traductions-et-sciences-ouvertes.html</link>
			<description></description>
			<pubDate>2024-10-02T14:34:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Colloques-conferences-et-debats/une-babel-des-sciences-traductions-et-sciences-ouvertes.html</guid>
		</item>
		<item>
			<title>Michael Hall, lauréat de la Grande médaille de l’Académie des sciences</title>
			<link>http://www.academie-sciences.fr/fr/Actualites-scientifiques/seance-2024-de-remise-de-la-grande-medaille-de-l-academie-des-sciences.html</link>
			<description></description>
			<pubDate>2024-09-17T12:06:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Actualites-scientifiques/seance-2024-de-remise-de-la-grande-medaille-de-l-academie-des-sciences.html</guid>
		</item>
		<item>
			<title>La Médaille d'Or du CNRS remise à Edith Heard </title>
			<link>http://www.academie-sciences.fr/fr/Communiques-de-presse/la-medaille-d-or-du-cnrs-remise-a-edith-heard.html</link>
			<description></description>
			<pubDate>2024-10-02T16:44:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Communiques-de-presse/la-medaille-d-or-du-cnrs-remise-a-edith-heard.html</guid>
		</item>
		<item>
			<title>Section Biologie humaine et sciences médicales</title>
			<link>http://www.academie-sciences.fr/fr/Listes-par-section-scientifique/section-biologie-humaine-et-sciences-medicales.html</link>
			<description></description>
			<pubDate>2015-01-12T12:54:51+01:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Listes-par-section-scientifique/section-biologie-humaine-et-sciences-medicales.html</guid>
		</item>
		<item>
			<title>Marc Fontecave, lauréat de l'ENI Energy Transition Award 2024</title>
			<link>http://www.academie-sciences.fr/fr/Membres-a-la-une/marc-fontecave-laureat-de-l-energy-transition-award-2024.html</link>
			<description></description>
			<pubDate>2024-09-19T12:18:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Membres-a-la-une/marc-fontecave-laureat-de-l-energy-transition-award-2024.html</guid>
		</item>
		<item>
			<title>Séance de remise de la Grande médaille de l’Académie des sciences 2022 à Terence Tao</title>
			<link>http://www.academie-sciences.fr/fr/Colloques-conferences-et-debats/ceremonie-grande-medaille-2022-terence-tao.html</link>
			<description></description>
			<pubDate>2023-03-21T13:06:00+01:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Colloques-conferences-et-debats/ceremonie-grande-medaille-2022-terence-tao.html</guid>
		</item>
		<item>
			<title>Appel à candidature : Grand Prix Lamonica de neurologie 2024</title>
			<link>http://www.academie-sciences.fr/fr/Appel-a-candidature/prix-lamonica-neurologie.html</link>
			<description></description>
			<pubDate>2024-07-19T08:00:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Appel-a-candidature/prix-lamonica-neurologie.html</guid>
		</item>
		<item>
			<title>Appel à candidature : Grand prix Sicard – Académie des sciences 2024</title>
			<link>http://www.academie-sciences.fr/fr/Appel-a-candidature/appel-a-candidature-grand-prix-sicard-academie-des-sciences-2024.html</link>
			<description></description>
			<pubDate>2024-07-29T09:47:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Appel-a-candidature/appel-a-candidature-grand-prix-sicard-academie-des-sciences-2024.html</guid>
		</item>
		<item>
			<title>Michèle Vergne, lauréate de la médaille Weyl-Wigner</title>
			<link>http://www.academie-sciences.fr/fr/Membres-a-la-une/michele-vergne-laureate-de-la-medaille-weyl-wigner.html</link>
			<description></description>
			<pubDate>2024-08-20T12:18:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Membres-a-la-une/michele-vergne-laureate-de-la-medaille-weyl-wigner.html</guid>
		</item>
		<item>
			<title>Pierre Tiollais</title>
			<link>http://www.academie-sciences.fr/fr/In-memoriam/hommage-a-pierre-tiollais.html</link>
			<description></description>
			<pubDate>2024-08-05T12:06:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/In-memoriam/hommage-a-pierre-tiollais.html</guid>
		</item>
		<item>
			<title>Section Biologie intégrative</title>
			<link>http://www.academie-sciences.fr/fr/Listes-par-section-scientifique/section-biologie-integrative.html</link>
			<description></description>
			<pubDate>2015-01-12T12:54:51+01:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Listes-par-section-scientifique/section-biologie-integrative.html</guid>
		</item>
		<item>
			<title>Campagne d'appels à candidature</title>
			<link>http://www.academie-sciences.fr/fr/Prix-et-medailles/campagne-d-appels-a-candidatures.html</link>
			<description></description>
			<pubDate>2014-07-03T14:34:15+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Prix-et-medailles/campagne-d-appels-a-candidatures.html</guid>
		</item>
		<item>
			<title>Mini-organes et embryons précoces in vitro : Quels enjeux ?</title>
			<link>http://www.academie-sciences.fr/fr/Colloques-conferences-et-debats/mini-organes-et-embryons-precoces-in-vitro-quels-enjeux.html</link>
			<description></description>
			<pubDate>2024-11-12T13:06:00+01:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Colloques-conferences-et-debats/mini-organes-et-embryons-precoces-in-vitro-quels-enjeux.html</guid>
		</item>
		<item>
			<title>Appel à candidature : prix Irène Joliot-Curie 2024</title>
			<link>http://www.academie-sciences.fr/fr/Appel-a-candidature/appel-a-candidature-prix-irene-joliot-curie.html</link>
			<description></description>
			<pubDate>2024-07-19T08:47:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Appel-a-candidature/appel-a-candidature-prix-irene-joliot-curie.html</guid>
		</item>
		<item>
			<title>Appel à candidature : Grand Prix Lamonica de cardiologie 2024</title>
			<link>http://www.academie-sciences.fr/fr/Appel-a-candidature/prix-lamonica-cardiologie.html</link>
			<description></description>
			<pubDate>2024-07-19T08:00:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Appel-a-candidature/prix-lamonica-cardiologie.html</guid>
		</item>
		<item>
			<title>Comité des Scientifiques pour l'Ukraine (CSU)</title>
			<link>http://www.academie-sciences.fr/fr/Comites-thematiques/comite-scientifiques-ukraine.html</link>
			<description></description>
			<pubDate>2022-04-01T09:30:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Comites-thematiques/comite-scientifiques-ukraine.html</guid>
		</item>
		<item>
			<title>Stéphane Mallat, IEEE Fourier Award for Signal Processing</title>
			<link>http://www.academie-sciences.fr/fr/Membres-a-la-une/stephane-mallat-ieee-fourier-award-for-signal-processing.html</link>
			<description></description>
			<pubDate>2024-04-29T12:00:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Membres-a-la-une/stephane-mallat-ieee-fourier-award-for-signal-processing.html</guid>
		</item>
		<item>
			<title>Lydéric Bocquet, lauréat de la médaille de l’innovation du CNRS 2024</title>
			<link>http://www.academie-sciences.fr/fr/Membres-a-la-une/lyderic-bocquet-laureat-de-la-medaille-de-l-innovation-du-cnrs-2024.html</link>
			<description></description>
			<pubDate>2024-07-09T12:18:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Membres-a-la-une/lyderic-bocquet-laureat-de-la-medaille-de-l-innovation-du-cnrs-2024.html</guid>
		</item>
		<item>
			<title>L’infiniment petit, avenir de la transition énergétique - Lydéric Bocquet - L'HUMANITÉ MAGAZINE</title>
			<link>http://www.academie-sciences.fr/fr/Articles-en-partenariat/l-infiniment-petit-avenir-de-la-transition-energetique-lyderic-bocquet-l-humanite-magazine.html</link>
			<description></description>
			<pubDate>2024-07-04T07:10:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Articles-en-partenariat/l-infiniment-petit-avenir-de-la-transition-energetique-lyderic-bocquet-l-humanite-magazine.html</guid>
		</item>
		<item>
			<title>Communiqué de presse | Dimanche, la vigilance républicaine s’impose</title>
			<link>http://www.academie-sciences.fr/fr/Communiques-de-presse/communique-de-presse-dimanche-la-vigilance-republicaine-s-impose.html</link>
			<description></description>
			<pubDate>2024-07-03T10:25:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Communiques-de-presse/communique-de-presse-dimanche-la-vigilance-republicaine-s-impose.html</guid>
		</item>
		<item>
			<title>Patrick Flandrin, lauréat du prix de l'éducation Carl Friedrich Gauss 2023</title>
			<link>http://www.academie-sciences.fr/fr/Membres-a-la-une/patrick-flandrin-laureat-du-prix-de-l-education-carl-friedrich-gauss-2023.html</link>
			<description></description>
			<pubDate>2024-04-29T12:00:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Membres-a-la-une/patrick-flandrin-laureat-du-prix-de-l-education-carl-friedrich-gauss-2023.html</guid>
		</item>
		<item>
			<title>Un poisson pour comprendre les cellules souches du cerveau - Laure Bally-Cuif - L'HUMANITÉ MAGAZINE</title>
			<link>http://www.academie-sciences.fr/fr/Articles-en-partenariat/un-poisson-pour-comprendre-les-cellules-souches-du-cerveau-laure-bally-cuif-l-humanite-magazine.html</link>
			<description></description>
			<pubDate>2024-06-06T07:10:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Articles-en-partenariat/un-poisson-pour-comprendre-les-cellules-souches-du-cerveau-laure-bally-cuif-l-humanite-magazine.html</guid>
		</item>
		<item>
			<title>Patrick Couvreur, lauréat de la &quot;Biomaterialia Golden Medal » et le « Controlled Release Society Founder Award » 2024</title>
			<link>http://www.academie-sciences.fr/fr/Membres-a-la-une/patrick-couvreur-laureat-de-la-biomaterialia-golden-medal-2024.html</link>
			<description></description>
			<pubDate>2024-05-28T12:00:00+02:00</pubDate>
			<guid isPermaLink="false">http://www.academie-sciences.fr/fr/Membres-a-la-une/patrick-couvreur-laureat-de-la-biomaterialia-golden-medal-2024.html</guid>
		</item>
	</channel>
</rss>

```

</detail>

Alkarex committed:
Merge branch 'edge' into better-bad-guids

Alkarex commented:
Another example from https://www.php.net/feed.atom
5 invalid GUIDs out of 741 (why!?) entries

```xml
<entry xmlns="http://www.w3.org/2005/Atom" xml:base="entries/2021-12-17-1.xml">
  <title>PHP 8.1.1 Released!</title>
  <id>https://www.php.net/archive/2021.php#2021-12-17-1</id>
  <published>2021-12-17T17:31:25+01:00</published>
  <updated>2021-12-17T17:31:25+01:00</updated>
  <link href="https://www.php.net/index.php#id2021-12-17-1" rel="alternate" type="text/html"/>
  <link href="https://www.php.net/archive/2021.php#2021-12-17-1" rel="via" type="text/html"/>
  <category term="releases" label="New PHP release"/>
  <category term="frontpage" label="PHP.net frontpage news"/>
  <content type="xhtml">
    <div xmlns="http://www.w3.org/1999/xhtml"><p>The PHP development team announces the immediate availability of PHP 8.1.1. This is a bug fix release.</p>

<p>All PHP 8.1 users are encouraged to upgrade to this version.</p>

<p>For source downloads of PHP 8.1.1 please visit our <a href="https://www.php.net/downloads.php">downloads page</a>,
Windows source and binaries can be found on <a href="https://windows.php.net/download/">windows.php.net/download/</a>.
The list of changes is recorded in the <a href="https://www.php.net/ChangeLog-8.php#8.1.1">ChangeLog</a>.
</p>    </div>
  </content>
</entry>
  <entry xmlns="http://www.w3.org/2005/Atom" xml:base="entries/2021-12-16-2.xml">
  <title>PHP 8.0.14 Released!</title>
  <id>https://www.php.net/archive/2021.php#2021-12-17-1</id>
  <published>2021-12-16T23:33:54+00:00</published>
  <updated>2021-12-16T23:33:54+00:00</updated>
  <link href="https://www.php.net/index.php#id2021-12-16-2" rel="alternate" type="text/html"/>
  <link href="https://www.php.net/archive/2021.php#2021-12-16-2" rel="via" type="text/html"/>
  <category term="releases" label="New PHP release"/>
  <category term="frontpage" label="PHP.net frontpage news"/>
  <content type="xhtml">
    <div xmlns="http://www.w3.org/1999/xhtml"><p>The PHP development team announces the immediate availability of PHP 8.0.14. This is a bug fix release.</p>

<p>All PHP 8.0 users are encouraged to upgrade to this version.</p>

<p>For source downloads of PHP 8.0.14 please visit our <a href="https://www.php.net/downloads.php">downloads page</a>,
Windows source and binaries can be found on <a href="https://windows.php.net/download/">windows.php.net/download/</a>.
The list of changes is recorded in the <a href="https://www.php.net/ChangeLog-8.php#8.0.14">ChangeLog</a>.
</p>    </div>
  </content>
</entry>
```

Frenzie commented:
Strange, I wonder how that happened. The 8.0.14 is actually `https://www.php.net/archive/2021.php#2021-12-16-2`. A human copy/paste mistake?

Edit: looks like I'm wrong(ish) :-) The ID is `2021-12-17-1` so the link/id in the feed is correct, but in the `href` it gives `2021-12-16-2`.
```html
<a id="2021-12-17-1" href="https://www.php.net/archive/2021.php#2021-12-16-2" rel="bookmark" class="bookmark">PHP 8.0.14 Released!</a>
```

math-GH commented on app/i18n/es/sub.php:
```diff
@@ -228,6 +228,16 @@
+			'_' => 'Article unicity criteria',	// TODO
+			'forced' => '<span title="Block the unicity criteria, even when the feed has duplicate articles">forced</span>',	// TODO
+			'help' => 'Relevant for invalid feeds.<br />⚠️ Changing the policy will create duplicates.',	// TODO
+			'id' => 'Standard ID (default)',	// TODO
```
@Alkarex 
I need some understanding for `Standard ID`:
Is it the GUID (Global Unique ID)?

Alkarex replied:
Yes, except that it is not a GUID but a unique ID within a given feed, and varies depending on the version of RSS / Atom.

See:
* RSS 2.0 `guid` https://www.rssboard.org/rss-specification#ltguidgtSubelementOfLtitemgt
* Atom `id` https://www.rfc-editor.org/rfc/rfc4287#page-19
* RSS 1.0 item URI https://web.resource.org/rss/1.0/spec#s5.5

math-GH replied:
Would `ID (default)` also be a good String?
It is difficult to translate `Standard` and `default` in the same String because both could be translated to something like `Standard ID (Standardeinstellung)`. Too much `Standard` in 1 String :D

Alkarex replied:
Yes, I think this is fine enough
